--- configure.ac~	2009-09-29 05:41:32.000000000 +0200
+++ configure.ac	2009-11-28 18:36:42.000000000 +0100
@@ -112,7 +112,7 @@
 dnl **************************************************************
 dnl Check for D-Bus GLib
 dnl **************************************************************
-PKG_CHECK_MODULES(DBUS_GLIB, dbus-glib-1)
+PKG_CHECK_MODULES(DBUS_GLIB, dbus-glib-1 gthread-2.0)
 
 dnl **************************************************************
 dnl Check for MozWebView
